base/dcp_sharded.go

```diff
@@ -46,11 +61,11 @@ func StartShardedDCPFeed(dbName string, configGroup string, uuid string, heartbe
 		return nil, err
 	}
 
-	dcpContextID := MD(spec.BucketName).Redact() + "-" + DCPImportFeedID
-	cbgtContext.loggingCtx = context.WithValue(context.Background(), LogContextKey{},
-		LogContext{CorrelationID: dcpContextID},
-	)
-
+	if cbgtContext.LithiumCompat.IsTrue() {
+		if spec.Scope != nil || spec.Collection != nil {
```

Comment:
I don't think StartShardedDCPFeed is running in any kind of retry loop, is it?  Given that, I don't see the value in persisting the LithiumCompat to the cbgtContext, if we're only ever going to error out right away - could just return as an error from initCBGTManager?

Reply:
That's a fair point - in theory we can do the `getMinNodeCompatVersion` check before starting the manager as it can be done with just a `cbgt.Cfg`. In theory there's the risk that this becomes a TOCTOU (i.e. a node that doesn't support collections joins the cluster in between the `getMinNodeCompatVersion` check and the manager startup), but IMHO this is a rare enough case that it's not worth worrying about.